commit 31c913191db3c680e1b71547e8d46713103ee805
Author: tangwenji <tang.wenji@zte.com.cn>
Date:   Wed Aug 16 16:39:00 2017 +0800

    target: fix null pointer regression in core_tmr_drain_tmr_list
    
    commit 88fb2fa7db7510bf1078226ab48d162d9854f3d4 upstream.
    
    The target system kernel crash when the initiator executes
    the sg_persist -A command,because of the second argument to
    be set to NULL when core_tmr_lun_reset is called in
    core_scsi3_pro_preempt function.
    
    This fixes a regression originally introduced by:
    
      commit 51ec502a32665fed66c7f03799ede4023b212536
      Author: Bart Van Assche <bart.vanassche@sandisk.com>
      Date:   Tue Feb 14 16:25:54 2017 -0800
    
          target: Delete tmr from list before processing
    
    Signed-off-by: tangwenji <tang.wenji@zte.com.cn>
    Signed-off-by: Nicholas Bellinger <nab@linux-iscsi.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/target/target_core_tmr.c b/drivers/target/target_core_tmr.c
index e22847bd79b9..61909b23e959 100644
--- a/drivers/target/target_core_tmr.c
+++ b/drivers/target/target_core_tmr.c
@@ -217,7 +217,8 @@ static void core_tmr_drain_tmr_list(
 	 * LUN_RESET tmr..
 	 */
 	spin_lock_irqsave(&dev->se_tmr_lock, flags);
-	list_del_init(&tmr->tmr_list);
+	if (tmr)
+		list_del_init(&tmr->tmr_list);
 	list_for_each_entry_safe(tmr_p, tmr_pp, &dev->dev_tmr_list, tmr_list) {
 		cmd = tmr_p->task_cmd;
 		if (!cmd) {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1112_31c913191db3c680e1b71547e8d46713103ee805.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1112_31c913191db3c680e1b71547e8d46713103ee805.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
